sdk/storage/storage-file/review/storage-file.api.ts

```diff
@@ -0,0 +1,787 @@
+  downloadToBuffer(buffer: Buffer, offset: number, count?: number, options?: DownloadFromAzureFileOptions): Promise<void>;
+  downloadToFile(filePath: string, offset?: number, count?: number, options?: FileDownloadOptions): Promise<Models.FileDownloadResponse>;
+  // (undocumented)
+  readonly filePath: string;
+  forceCloseHandle(handleId: string, options?: FileForceCloseHandlesOptions): Promise<Models.FileForceCloseHandlesResponse>;
+  forceCloseHandlesSegment(marker?: string, options?: FileForceCloseHandlesOptions): Promise<Models.FileForceCloseHandlesResponse>;
+  getProperties(options?: FileGetPropertiesOptions): Promise<Models.FileGetPropertiesResponse>;
+  getRangeList(options?: FileGetRangeListOptions): Promise<FileGetRangeListResponse>;
+  listHandlesSegment(marker?: string, options?: FileListHandlesSegmentOptions): Promise<Models.FileListHandlesResponse>;
+  resize(length: number, options?: FileResizeOptions): Promise<Models.FileSetHTTPHeadersResponse>;
+  setHTTPHeaders(fileHTTPHeaders?: FileHTTPHeaders, options?: FileSetHTTPHeadersOptions): Promise<Models.FileSetHTTPHeadersResponse>;
+  setMetadata(metadata?: Metadata, options?: FileSetMetadataOptions): Promise<Models.FileSetMetadataResponse>;
+  setProperties(properties?: FileProperties): Promise<SetPropertiesResponse>;
+  // (undocumented)
+  readonly shareName: string;
+  startCopyFromURL(copySource: string, options?: FileStartCopyOptions): Promise<Models.FileStartCopyResponse>;
+  uploadBrowserData(browserData: Blob | ArrayBuffer | ArrayBufferView, options?: UploadToAzureFileOptions): Promise<void>;
+  uploadFile(filePath: string, options?: UploadToAzureFileOptions): Promise<void>;
+  uploadRange(body: HttpRequestBody, offset: number, contentLength: number, options?: FileUploadRangeOptions): Promise<Models.FileUploadRangeResponse>;
+  uploadRangeFromURL(sourceURL: string, sourceOffset: number, destOffset: number, count: number, options?: FileUploadRangeFromURLOptions): Promise<Models.FileUploadRangeFromURLResponse>;
+  uploadResetableStream(streamFactory: (offset: number, count?: number) => NodeJS.ReadableStream, size: number, options?: UploadToAzureFileOptions): Promise<void>;
+  UploadSeekableBlob(blobFactory: (offset: number, size: number) => Blob, size: number, options?: UploadToAzureFileOptions): Promise<void>;
+  uploadStream(stream: Readable, size: number, bufferSize: number, maxBuffers: number, options?: UploadStreamToAzureFileOptions): Promise<void>;
+}
+
+// @public
+interface FileCreateOptions extends FileAndDirectoryCreateCommonOptions, CommonOptions {
+  abortSignal?: AbortSignalLike;
+  fileHTTPHeaders?: FileHTTPHeaders;
+  metadata?: Metadata;
+}
+
+// @public
+interface FileDeleteOptions extends CommonOptions {
+  abortSignal?: AbortSignalLike;
+}
+
+// @public
+interface FileDownloadOptions extends CommonOptions {
+  abortSignal?: AbortSignalLike;
+  maxRetryRequests?: number;
```

Comment:
according to the comments certain errors doesn't trigger retry policy
```
   * How many retries will perform when original body download stream unexpected ends.
   * Above kind of ends will not trigger retry policy defined in a pipeline,
   * because they doesn't emit network errors.
```

Reply:
Yes. `maxRetryRequests` is different from `maxRetryCount` defined in retry policy.

Retry policy works when request has network error or storage server error **before starting downloading response stream**.

This means, when downloading a 1TB file, if network broken after 500GB, retry policy won't be triggerred.

The retry defined here covers this scenario and defines how maxRetryRequests will happen.


---
In reply to: [334126212](https://github.com/Azure/azure-sdk-for-js/pull/5477#discussion_r334126212) [](ancestors = 334126212)

Reply:
will consider.

Reply:
Check other languages.

Reply:
#6019